R/deploymentTarget.R

```diff
@@ -91,7 +91,13 @@ findDeploymentTarget <- function(
   # Otherwise, identify a target account (given just one available or prompted
   # by the user), generate a name, and locate the deployment.
   accountDetails <- findAccountInfo(account, server, error_call = error_call)
-  appName <- generateAppName(appTitle, recordPath, accountDetails$name, unique = FALSE)
+  appTitle <- generateTitle(recordPath = recordPath, title = appTitle)
```

Comment:
This has the side-effect of generating an `appTitle` for shinyapps.io content, which does not support title. Alternatively, we could generate a title only for Connect deployments.

<img width="199" alt="image" src="https://github.com/rstudio/rsconnect/assets/362187/6b39b337-803f-46dd-ad96-fbf83d676132">

The other drawback of this approach is that as soon as we have generated the title, we lose track of the fact that the title was not user-specified. This means that the generated title will always overwrite the deployment record title.

Reply:
We may want to distinguish between "user provided title" and "generated title" when updating a discovered deployment record.

```r
title <- userTitle %||% deployment$title %||% generatedTitle
```

Reply:
After more testing, we probably want to avoid generating titles for shinyapps.io.

The IDE deployment pane forces a normalized "title", which becomes the title and name:

<img width="270" alt="image" src="https://github.com/rstudio/rsconnect/assets/362187/47fbfeb3-a899-4457-9b9f-c682402d297c">

After manually normalizing the title with "incredible-shiny-application" and completing this workflow, the deployment record name and title both receive the same value (as forced by the IDE).

Reply:
Using the CRAN version of `rsconnect::deployApp()` leaves a shinyapps.io deployment record with `title=NULL`. The IDE deployment workflow creates deployment records with title and name both with the same value.

The IDE allows mixed-case names(titles) for shinyapps.io, but mixed-case is not produced by `generateAppName()`.